[homekit] Bad client credentials

This is to re-open : https://github.com/openhab/openhab2-addons/issues/2798

```
11:03:39.632 [ERROR] [wulfe.hap.impl.pairing.PairingManager] - Exception encountered while processing pairing request com.nimbusds.srp6.SRP6Exception: Bad client credentials 
	at com.beowulfe.hap.impl.pairing.HomekitSRP6ServerSession.step2(HomekitSRP6ServerSession.java:331) [224:org.openhab.io.homekit:2.2.0] 
	at com.beowulfe.hap.impl.pairing.SrpHandler.step2(SrpHandler.java:76) [224:org.openhab.io.homekit:2.2.0] 
	at com.beowulfe.hap.impl.pairing.SrpHandler.handle(SrpHandler.java:47) [224:org.openhab.io.homekit:2.2.0] 
	at com.beowulfe.hap.impl.pairing.PairingManager.handle(PairingManager.java:44) [224:org.openhab.io.homekit:2.2.0] 
	at com.beowulfe.hap.impl.connections.HttpSession.handlePairSetup(HttpSession.java:107) [224:org.openhab.io.homekit:2.2.0] 
	at com.beowulfe.hap.impl.connections.HttpSession.handleRequest(HttpSession.java:51) [224:org.openhab.io.homekit:2.2.0] 
	at com.beowulfe.hap.impl.connections.ConnectionImpl.doHandleRequest(ConnectionImpl.java:51) [224:org.openhab.io.homekit:2.2.0] 
	at com.beowulfe.hap.impl.connections.ConnectionImpl.handleRequest(ConnectionImpl.java:46) [224:org.openhab.io.homekit:2.2.0] 
	at com.beowulfe.hap.impl.http.impl.AccessoryHandler.channelRead0(AccessoryHandler.java:47) [224:org.openhab.io.homekit:2.2.0] 
	at com.beowulfe.hap.impl.http.impl.AccessoryHandler.channelRead0(AccessoryHandler.java:15) [224:org.openhab.io.homekit:2.2.0] 
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105) [224:org.openhab.io.homekit:2.2.0] 
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308) [224:org.openhab.io.homekit:2.2.0] 
	at io.netty.channel.AbstractChannelHandlerContext.access$600(AbstractChannelHandlerContext.java:32) [224:org.openhab.io.homekit:2.2.0] 
	at io.netty.channel.AbstractChannelHandlerContext$7.run(AbstractChannelHandlerContext.java:299) [224:org.openhab.io.homekit:2.2.0] 
	at io.netty.util.concurrent.DefaultEventExecutor.run(DefaultEventExecutor.java:36) [224:org.openhab.io.homekit:2.2.0] 
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112) [224:org.openhab.io.homekit:2.2.0] 
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137) [224:org.openhab.io.homekit:2.2.0] 
	at java.lang.Thread.run(Thread.java:745) [?:?]
```

I keep getting the same error.

Please note these messages below which might be interesting for debugging purposes:
Entering second stage of pair for openHAB
Bad client credentials
... and something with Unauthorized connection keep alive, which I do not know what is.

Here is a subset of my trace log:

2019-06-05 22:18:03.274 [INFO ] [.hap.impl.http.impl.AccessoryHandler] - Terminated homekit connection from /192.168.1.242:49705

2019-06-05 22:18:03.206 [DEBUG] [ulfe.hap.impl.pairing.PairingManager] - Entering second stage of pair for openHAB
2019-06-05 22:18:03.256 [ERROR] [ulfe.hap.impl.pairing.PairingManager] - Exception encountered while processing pairing request
com.nimbusds.srp6.SRP6Exception: Bad client credentials
        at com.beowulfe.hap.impl.pairing.HomekitSRP6ServerSession.step2(HomekitSRP6ServerSession.java:270) ~[191:org.openhab.io.homekit:2.5.0.201902270358]
        at com.beowulfe.hap.impl.pairing.SrpHandler.step2(SrpHandler.java:76) ~[191:org.openhab.io.homekit:2.5.0.201902270358]
        at com.beowulfe.hap.impl.pairing.SrpHandler.handle(SrpHandler.java:47) ~[191:org.openhab.io.homekit:2.5.0.201902270358]
        at com.beowulfe.hap.impl.pairing.PairingManager.handle(PairingManager.java:44) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at com.beowulfe.hap.impl.connections.HttpSession.handlePairSetup(HttpSession.java:109) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at com.beowulfe.hap.impl.connections.HttpSession.handleRequest(HttpSession.java:53) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at com.beowulfe.hap.impl.connections.ConnectionImpl.doHandleRequest(ConnectionImpl.java:54) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at com.beowulfe.hap.impl.connections.ConnectionImpl.handleRequest(ConnectionImpl.java:47) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at com.beowulfe.hap.impl.http.impl.AccessoryHandler.channelRead0(AccessoryHandler.java:52) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at com.beowulfe.hap.impl.http.impl.AccessoryHandler.channelRead0(AccessoryHandler.java:17) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at io.netty.channel.AbstractChannelHandlerContext.access$600(AbstractChannelHandlerContext.java:32) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at io.netty.channel.AbstractChannelHandlerContext$7.run(AbstractChannelHandlerContext.java:299) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at io.netty.util.concurrent.DefaultEventExecutor.run(DefaultEventExecutor.java:36) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137) [191:org.openhab.io.homekit:2.5.0.201902270358]
        at java.lang.Thread.run(Thread.java:748) [?:?]
2019-06-05 22:18:03.258 [INFO ] [ap.impl.http.HomekitClientConnection] - 401 /pair-setup
2019-06-05 22:18:03.259 [TRACE] [pl.http.impl.NettyHomekitHttpService] - WRITE UnpooledUnsafeDirectByteBuf(ridx: 0, widx: 72, cap: 256) [/192.168.1.242:49705]:
00000000 48 54 54 50 2F 31 2E 31 20 34 30 31 20 55 6E 61 HTTP/1.1 401 Una
00000010 75 74 68 6F 72 69 7A 65 64 0D 0A 43 6F 6E 74 65 uthorized..Conte
00000020 6E 74 2D 4C 65 6E 67 74 68 3A 20 30 0D 0A 43 6F nt-Length: 0..Co
00000030 6E 6E 65 63 74 69 6F 6E 3A 20 6B 65 65 70 2D 61 nnection: keep-a
00000040 6C 69 76 65 0D 0A 0D 0A                         live....

--- a/bundles/org.openhab.io.homekit/README.md
+++ b/bundles/org.openhab.io.homekit/README.md
@@ -17,6 +17,7 @@ This will be used in iOS when pairing. The pin code is in the form "###-##-###".
 Requirements beyond this are not clear, and Apple enforces limitations on eligible pins within iOS.
 At the very least, you cannot use repeating (111-11-111) or sequential (123-45-678) pin codes.
 If your home network is secure, a good starting point is the pin code used in most sample applications: 031-45-154.
+Check for typos in the pin-code if you encounter "Bad Client Credential" errors during pairing.  
 
 Other settings, such as using Fahrenheit temperatures, customizing the thermostat heat/cool/auto modes, and specifying the interface to advertise the HomeKit bridge (which can be edited in Paper UI standard mode) are also illustrated in the following sample:
 
@@ -60,7 +61,7 @@ org.openhab.homekit:maximumTemperature=100
 After setting this global configuration, you will need to tag your [openHAB items](https://www.openhab.org/docs/configuration/items.html) for HomeKit in order to map them to an ontology.
 For our purposes, you may consider HomeKit accessories to be of two forms: simple and complex.
 
-A simple accessory will be mapped to a single openHAB item (i.e. a Lighbulb is mapped to a Switch, Dimmer, or Color item).
+A simple accessory will be mapped to a single openHAB item (i.e. a Lightbulb is mapped to a Switch, Dimmer, or Color item).
 A complex accessory will be made up of multiple openHAB items (i.e. a Thermostat is composed of Heating and Cooling thresholds, a mode, and current temperature).
 Complex accessories require a tag on a Group indicating the accessory type, as well as tags on the items it composes.
 

